resolwe_bio/processes/WGBS/walt.yml

```diff
@@ -144,38 +166,58 @@
             -o "${NAME}_sorted.bam" \
             "${NAME}_mapped.bam"
       re-checkrc "Samtools sort command failed."
-      
-      samtools index "${NAME}_sorted.bam" "${NAME}_sorted.bam.bai"
-      re-checkrc "Samtools index command failed."
 
-      to-mr -o "${NAME}.mr" -m general "${NAME}_mapped.bam"
+      to-mr -o "${NAME}_mapped.mr" -m general "${NAME}_mapped.bam"
       re-checkrc "Conversion of bam to mr failed."
 
       LC_ALL=C \
       sort \
         -k 1,1 -k 2,2n -k 3,3n -k 6,6 \
         -S{{ requirements.resources.memory }}M \
         -o "${NAME}_sorted.mr" \
-        "${NAME}.mr"
+        "${NAME}_mapped.mr"
       re-progress 0.8
 
       {% if rm_dup %}
-        duplicate-remover -o "${NAME}.mr" "${NAME}_sorted.mr"
+        duplicate-remover -o "${NAME}_dremove.mr" "${NAME}_sorted.mr"
         re-checkrc "Removing duplicates failed."
       {% else %}
-        mv "${NAME}_sorted.mr" "${NAME}.mr"
+        mv "${NAME}_sorted.mr" "${NAME}_dremove.mr"
+      {% endif %}
+
+      {% if spikein_options.spikein_name is defined %}
+        grep ^{{ spikein_options.spikein_name }} "${NAME}_dremove.mr" > "${NAME}_spikein.mr"
+        re-checkrc "Given unmethylated control sequence name was not found."
```

Comment:
If no reads align the the given `spikein_name`, the grep command fails, since there are no lines with `spikein_name` in the mr file.

Reply:
Yes, at first I intended to include another if statement that would check if a given spikein name is present in the .mr file, but this would hurt the performance when working with large files. Should I include this regardless?

Reply:
If you ommit the `re-checkrc`, the process passes OK. The issue is just that the user doesn't know that nothing was removed from the .mr file (e.g. typo in the spike-in name). At least a warning should be in place that notifies the user of this situation